services/discord_bot_service.js

```diff
@@ -0,0 +1,93 @@
+const { discordBot } = require('../discordBot.js');
+const userService = require('../services/user_service');
+const repo = require('../repositories/discord_bot_repository');
+
+async function generateInviteLink(userId) {
+  try {
+    const guild = await getGuild(discordBot.discordClient, discordBot.guildId);
+    const channel = await getChannel(guild, discordBot.channelId);
+    const invite = await createInvite(channel); 
+    if (userId !== null) {
```

Comment:
I think at this point we should we pretty sure userId is not null and if for some reason it is, we should let the database fail